static/app/utils/api/useQuery.stories.tsx

```diff
@@ -0,0 +1,343 @@
+    const interval = setInterval(() => setSeconds(prev => prev + 1), 1_000);
+    return () => clearInterval(interval);
+  }, []);
+
+  return (
+    <div>
+      Loading...
+      <p>Rendered for {seconds} seconds</p>
+    </div>
+  );
+}
+
+function ToggleMounted({
+  children,
+  queryKeys,
+}: {
+  children: ReactNode;
+  queryKeys: Array<readonly unknown[]>;
+}) {
+  const [isMounted, setIsMounted] = useState(true);
+  const queryClient = useQueryClient();
+
+  return (
+    <Fragment>
+      <Flex align="flex-start" column gap={space(1)}>
+        <ButtonBar merged>
+          <Button size="sm" onClick={() => setIsMounted(prev => !prev)}>
+            {isMounted ? 'Unmount' : 'Mount'}
+          </Button>
+          <Button
+            size="sm"
+            onClick={() =>
+              queryKeys.forEach(queryKey =>
+                queryClient.resetQueries({queryKey, exact: true})
+              )
+            }
+          >
+            Reset Queries
+          </Button>
+        </ButtonBar>
+        <SizingWindow>{isMounted ? children : null}</SizingWindow>
+      </Flex>
+    </Fragment>
+  );
+}
+
+function DataContainer({
+  children,
+  options,
+}: {
+  options: UseQueryOptions;
+  children?: ReactNode;
+}) {
+  const {
+    data,
+    error,
+    isError,
+    isFetching,
+    isInitialLoading,
+    isLoading,
+    isSuccess,
+    refetch,
+    status,
+  } = useQuery(options);
+
+  return (
+    <Flex column gap={space(1)}>
+      {children}
+      <Button size="xs" onClick={() => refetch()}>
+        refetch
+      </Button>
+      <ObjectInspector
+        data={{
+          status,
+          isFetching,
+          isLoading,
+          isInitialLoading,
+          isError,
+          error,
+          isSuccess,
+          data,
+        }}
+        expandLevel={2}
+      />
+    </Flex>
+  );
+}
+
+export default storyBook('useQuery', story => {
+  story('README', () => (
+    <Fragment>
+      <p>
+        This is a set of examples for how to call <code>useQuery()</code> and make the
+        most of it's built-in fetching/loading/error/success states.
+      </p>
+      <p>
+        These examples import from <code>@tanstack/react-query</code> directly, instead of
+        from <code>sentry/utils/queryClient</code> in order to directly test the api
+        without sentry specific helpers getting in the way. What's being tested are the
+        return types, which are consistent between <code>sentry/utils/queryClient</code>{' '}
+        and <code>@tanstack/react-query</code>.
+      </p>
+      <p>
+        You should prefer to import from <code>sentry/utils/queryClient</code> as much as
+        possible to benefit from easier and consistent data fetching within sentry.
+      </p>
+    </Fragment>
+  ));
+  story('TL/DR', () => (
+    <Fragment>
+      <p>
+        It seems like you can get really far by checking BOTH{' '}
+        <code>isFetching || isLoading</code> when you are waiting for the first render,
```

Comment:
I not sure we should encourage checking for `isFetching || isLoading`, since you'll lose the benefits of background refetches if you do so

Reply:
i'll make more examples!